File: UIL/UIL/Indeed 2018/Indeed2018/src/python/hand.py
fh = open("judges/hand.dat")

lines = int(fh.readline().strip())
   

for line in range(0,lines):
   knights = int(fh.readline().strip())
   #print("knights =",knights)
   a = 1
   b = 1
   cnt = 0
   while True:
      c = a + b
      #print(a,'+',b,'=',c)
      if c > knights:
         break
      cnt = cnt + 1
      a = b
      b = c
   print(cnt)

File: UIL/UIL/Indeed 2018/Indeed2018/src/hand.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class hand 
{

	public static void main(String[] args) throws FileNotFoundException 
	{
		Scanner file = new Scanner(new File("hand.dat"));
		
		long cases = file.nextLong();
		while(cases-->0)
		{
			long number_Of_knights = file.nextLong();
			long a = 1;
			long b = 1;
			
			long cnt =0;
			
			while(true)
			{
				long c = a+b;
				
				if(c>number_Of_knights)
				{
					break;
				}
				cnt++;
				a=b;
				b=c;
			}
			
			System.out.println(cnt);
		}

	}

}
